### PR TITLE
Dpdk backend: Fix dpdk test infrastructure to enable comparison of error output

### PR DESCRIPTION
**Pull request by usha1830**
Fixes https://github.com/p4lang/p4c/issues/3640
The changes in this PR includes:
- New directories for dpdk errors, pna errors and psa errors.
- Update to the dpdk regression scripts for this change.
- Enabling comparison of error output (-error files for dpdk)

### CODE REVIEW

**Comment by hesingh** [comment]
Does -f work for dpdk successful and failed tests with this PR?

**Review by mbudiu-vmw — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
Should I merge this?

**Comment by usha1830** [comment]
> Does -f work for dpdk successful and failed tests with this PR?

Yes. I verified by manually running few tests.

**Comment by usha1830** [comment]
> Should I merge this?

@hesingh Any other comments/questions?

File: testdata/p4_16_dpdk_errors_outputs/pna-add-on-miss-err1.p4.bfrt.json [hunk 1]
```diff
@@ -0,0 +1,167 @@
+{
+  "schema_version" : "1.0.0",
```
Comment (jafingerhut) [inline testdata/p4_16_dpdk_errors_outputs/pna-add-on-miss-err1.p4.bfrt.json:2]: Do you want to require checking in expected .bfrt.json files for programs that you expect an error message for?  Do you expect to get much test coverage out of such expected output files?

File: testdata/p4_16_dpdk_errors_outputs/pna-dpdk-direct-meter-err-1.p4-error [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[--Werror=unexpected] error: execute method for different DirectMeter instances (meter0_0 and meter1_0) called within same action
+[--Werror=unexpected] error: execute method for different DirectMeter instances (meter0_0 and meter1_0) called within same action
+pna-dpdk-direct-meter-err-1.p4(116): [--Werror=target-error] error: Direct counters and direct meters are unsupported for wildcard match table ipv4_da_lpm
+    table ipv4_da_lpm {
+          ^^^^^^^^^^^
```
Comment (jafingerhut) [inline testdata/p4_16_dpdk_errors_outputs/pna-dpdk-direct-meter-err-1.p4-error:5]: Is this error message accurate?  That is, does DPDK restrict direct counters and meters to ONLY be used on tables where all key fields are match_kind exact?  If that is a restriction for the DPDK implementation, why?  It seems like a strange restriction to me.

**Review by jafingerhut — APPROVED** [review]
Looks reasonable to merge to me.  My comments are minor, and can be addressed later after merge, if it is determined they are significant comments (which they might not be).

File: testdata/p4_16_dpdk_errors_outputs/pna-dpdk-direct-meter-err-1.p4-error (hunk 2 above)
Comment (usha1830) [reply testdata/p4_16_dpdk_errors_outputs/pna-dpdk-direct-meter-err-1.p4-error:5]: Yes Andy. Currently, dpdk target supports direct meter and direct counter only for exact match tables (all keys with match kind exact). We will remove this check as and when the target support is added.

File: testdata/p4_16_dpdk_errors_outputs/pna-add-on-miss-err1.p4.bfrt.json (hunk 1 above)
Comment (usha1830) [reply testdata/p4_16_dpdk_errors_outputs/pna-add-on-miss-err1.p4.bfrt.json:2]: I think we can avoid this check. I will update

**Comment by hesingh** [comment]
> > Should I merge this?
> 
> @hesingh Any other comments/questions?

No, thanks. Good to go.

**Pull request merged**